include/fiction/io/write_svg_layout.hpp

```diff
@@ -37,6 +43,58 @@ struct write_qca_layout_svg_params
     bool simple = false;
 };
 
+/**
+ * Enumeration to specify the color mode for the SVG output.
+ */
+enum class color_mode : uint8_t
+{
+    /**
+     * Dark mode.
+     */
+    DARK,
+    /**
+     * Light mode.
+     */
+    LIGHT
+};
+/**
+ * Parameters for writing SiDB layouts to SVG format.
+ */
+struct write_sidb_layout_svg_params
+{
+    /**
+     * Size of the H-Si lattice points in SVG units.
+     */
+    double lattice_point_size = 0.3;
+    /**
+     * Size of the SiDB in SVG units.
+     */
+    double sidb_size = 0.9;
+    /**
+     * Border width of the SiDB.
+     */
+    double sidb_border_width = 0.3;
+    /**
+     * The color mode of the background for the SVG output.
+     */
+    color_mode color_background = color_mode::DARK;
+};
+
+/**
+ * Parameters for writing SVG layouts.
+ */
+struct write_svg_layout_params
+{
+    /**
+     * Parameters for writing SVG QCA layouts.
+     */
+    write_qca_layout_svg_params qca_params;
+    /**
+     * Parameters for writing SiDB layouts to SVG format.
+     */
+    write_sidb_layout_svg_params sidb_params;
+};
```

Comment:
Why do we need this? Does it make sense to group these two completely different parameter sets in one object?

Reply:
I would argue, each SVG function should have its own set of parameters.